drivers/gpio/CMakeLists.txt

```diff
@@ -18,6 +18,6 @@ zephyr_sources_ifdef(CONFIG_GPIO_QMSI       gpio_qmsi.c)
 zephyr_sources_ifdef(CONFIG_GPIO_QMSI_SS    gpio_qmsi_ss.c)
 zephyr_sources_ifdef(CONFIG_GPIO_SCH        gpio_sch.c)
 zephyr_sources_ifdef(CONFIG_GPIO_STM32      gpio_stm32.c)
-zephyr_sources_ifdef(CONFIG_GPIO_SAM0       gpio_sam0.c)
+zephyr_sources_ifdef(CONFIG_GPIO_SAM0       gpio_sam0.c gpio_sam0_eic.c)
```

Comment:
I'm not an expert on DTS but it feels a bit like a hack. Shouldn't EIC - being an independent driver - have its own enable option in DTS which is then set by the GPIO driver. The end effect would be exactly the same but the design somewhat cleaner. I.e. here in CMakeList they would be enabled independently and the dependency would be defined in DTS. Especially that gpio_sam0.c is written in a way that parts of the code dependent on EIC are not compiled if EIC is not enabled.

Reply:
I'm happy either way.  v1 mixed the drivers, v2 (which I never pushed) had the EIC as it's own driver with an ugly API, and v3 (this version) tightly couples them.